### DISCUSSION TITLE
Feature Suggestion: Clipboard Integration

### DISCUSSION THREAD

**Discussion by TobiasWallner** [jupyter-xeus/cpp-terminal#265] [Ideas]
To answer the qestion from @flagarde from issue #257:

> Have you found something missing you would like to be implemented?

Copy and pasting from and to the clipboard is something that is often needed when working with text in the console. For example when a key kompination like Ctrl+X/C/V is pressed. This would not generate a CopyPasteEvent in your library, so it may be nice to have some functions exposed that let me copy and paste to and from the clipboard.

I started with this already in my project and made pasting from the clipboard into my editor. 
see: https://github.com/TobiasWallner/Lime/blob/main/lime/Lime.cpp#L200-L288

Note that currently there is an issue on Linux. If the clipboard is empty, then a nullptr will be dereferenced. Have not gotten to look into it. (Linux version of the copy and paset requires X11 and windows just windows.h)

**Comment by flagarde** [comment]
Nice addition indeed but we should use the case with Wayland too which is the future... One day.. then this requires installing X11 that some computer don't have... 
Maybe loading the library on fly is possible ?

**Reply by TobiasWallner** [reply]
To be honest, I know way to little about Linux, I was happy to get it working somehow on my Debian Virtual Machine where X11 worked right out of the box. Getting the windows copy and paste to work was asonishingly straight forward and easy to understand
